src/services/passwordGeneration.service.ts

```diff
@@ -217,7 +223,82 @@ export class PasswordGenerationService implements PasswordGenerationServiceAbstr
             }
         }
 
-        return this.optionsCache;
+        const enforcedPolicyOptions = await this.getPasswordGeneratorPolicyOptions();
+        if (this.optionsCache.length < enforcedPolicyOptions.minLength) {
```

Comment:
Is there a way you can just return `null` here if there are no policies to enforce? Then you could just skip all of this condition checking.

Reply:
Yes - added a few lines to both this function and the `getPasswordGeneratorPolicyOptions` to handle that and keep the UI layer happy at the same time.